File: documentation-examples-kotlin/src/kotlin/dev/typr/foundations/docs/postgresql/JsonTypes.kt
package dev.typr.foundations.docs.postgresql

import dev.typr.kotlinfoundations.*
import dev.typr.kotlinfoundations.data.*
import dev.typr.kotlinfoundations.data.Json
import dev.typr.kotlinfoundations.data.Jsonb

@Suppress("unused")
class JsonTypes {
    //start
    val jsonType: PgType<Json> = PgTypes.json
    val jsonbType: PgType<Jsonb> = PgTypes.jsonb

    // Parse and use JSON
    val data: Json = Json("{\"name\": \"John\"}")
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/postgresql/JsonTypes.scala
package dev.typr.foundationssc.docs.postgresql
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*



@SuppressWarnings(Array("unused"))
object JsonTypes:
  //start
  val jsonType: PgType[Json] = PgTypes.json
  val jsonbType: PgType[Jsonb] = PgTypes.jsonb

  // Parse and use JSON
  val data: Json = new Json("{\"name\": \"John\"}")
  //stop
